overlord/snapstate/handlers.go

```diff
@@ -737,6 +738,37 @@ func (m *SnapManager) undoMountSnap(t *state.Task, _ *tomb.Tomb) error {
 	return m.backend.RemoveSnapDir(snapsup.placeInfo(), otherInstances)
 }
 
+// saveSnapDisabledServices queries for all currently disabled snap services and
+// saves those inside SideInfo for the provided snap state
+func (m *SnapManager) saveSnapDisabledServices(snapst *SnapState, pb progress.Meter) error {
+	info, err := snapst.CurrentInfo()
+	if err != nil {
+		return err
+	}
+
+	// save the list of services that are in the disabled state before unlinking
+	// and thus removing the snap services
+	snapSvcStates, err := m.backend.CurrentSnapServiceStates(info, pb)
+	if err != nil {
+		return err
+	}
+
+	// commit the disabled services state to the current side info
+	sideInfo := snapst.CurrentSideInfo()
+
+	// clear out the list of disabled services if it exists to repopulate fresh
+	sideInfo.DisabledServices = []string{}
+
+	// add all disabled services to the list
+	for svc, isEnabled := range snapSvcStates {
+		if !isEnabled {
+			sideInfo.DisabledServices = append(sideInfo.DisabledServices, svc)
```

Comment:
if this is needed only during the change, we should proceed like we do for the SnapPath in download-snap and store the information just for the change in SnapSetup, see

https://github.com/snapcore/snapd/blob/master/overlord/snapstate/handlers.go#L560

then that information is used in later tasks, example:

https://github.com/snapcore/snapd/blob/master/overlord/snapstate/handlers.go#L619